packages/Breadcrumbs/src/Breadcrumbs.js

```diff
@@ -11,14 +12,15 @@ import * as sc from "./Breadcrumbs.styles";
 function Breadcrumbs(props) {
   const { children, isDark, isAutoCollapsed, ...moreProps } = props;
   const I18n = useI18n();
-  const childrenCount = React.Children.count(children);
-  const shouldShowExpandButton = isAutoCollapsed && React.Children.count(children) > MAXIMUM_NUM_OF_LEVEL;
-  const hasOnlyOneChild = childrenCount === 1;
+
   const [isCollapsed, setIsCollapsed] = React.useState(shouldShowExpandButton);
+  const linkChildren = [extractChildren(children, ["Breadcrumbs.Link"])["Breadcrumbs.Link"]].flat();
+  const hasOnlyOneChild = linkChildren.length === 1;
+  const shouldShowExpandButton = isAutoCollapsed && linkChildren.length > MAXIMUM_NUM_OF_LEVEL;
 
   React.useLayoutEffect(() => {
     setIsCollapsed(shouldShowExpandButton);
-  }, [childrenCount, shouldShowExpandButton]);
+  }, [linkChildren.length, shouldShowExpandButton]);
```

Comment:
Why do we need an effect here in the first place? It looks like a derived initial state to me.

Reply:
I just want to recalculate the collapsed state when the number of children has been changed, like the example you mentioned above